### DISCUSSION TITLE
how to use include

### DISCUSSION THREAD

**Discussion by lewisblackburn** [MichalLytek/typegraphql-prisma#80] [Q&A] [answered]
![Screenshot 2021-03-16 at 11 50 44](https://user-images.githubusercontent.com/51877955/111304623-d9d2b280-864d-11eb-8afa-ede7f808e81f.png)

```js
  @Query(() => [Tweet], {
    nullable: false,
  })
  async tweets(
    @Ctx() ctx: Context,
    @Args() args: FindManyTweetArgs
  ): Promise<Tweet[]> {
    const tweets = await ctx.prisma.tweet.findMany({
      include: {
        author: true,
      },
    });
    console.log(tweets);
    return tweets;
  }
  ```
  
  I'm having trouble returning included data with GraphQL if anyone can help me out that would be great :) Sorry if this is the wrong place ask!

**Comment by MichalLytek** [comment, marked as answer]
Have you read the docs?
https://github.com/MichalLytek/typegraphql-prisma/blob/main/Readme.md

![image](https://user-images.githubusercontent.com/10618781/111308404-edd0e100-865a-11eb-84f0-58de6388e69a.png)

![image](https://user-images.githubusercontent.com/10618781/111308450-fb866680-865a-11eb-9f4b-1838afbad36e.png)

**Reply by lewisblackburn** [reply]
I read through the docs and couldn't find anything for prisma include, I would use TweetCrudResovler and TweetRelationsResolver but this causes the graphql n+1.

**Reply by MichalLytek** [reply]
@lewisblackburn Prisma has a built-in dataloader which prevents N+1 issues:
https://github.com/MichalLytek/typegraphql-prisma/issues/59#issuecomment-789687659

**Reply by lewisblackburn** [reply]
Thanks!

**Comment by 0xborisgregor** [comment]
for anyone who's using crud resolver with `mongodb` and find filter on relation not working: you must add `@db.ObjectId` to the foreign key property otherwise filter on relation will only return `null`
